### PR TITLE
Lmbd file handles not automatically deleted on Windows

### PR DESCRIPTION
**Pull request by hansieodendaal**
## Description
Added function `delete_db_from_scope` as extended implementation of `LMDBStore` to explicitly remove the LMDBStore struct from scope. This will cause LMBD to drop and all databases and close all db handles.

## Motivation and Context
Tests were failing on Windows due to test folder being deleted before database objects went out of scope. Files are handled differently on Linux and Windows.

## How Has This Been Tested?
Current test were extended to explicitly call the new `delete_db_from_scope` function before test folders are being deleted.
Tested on Windows 10 and Ubuntu Linux.

## Types of changes

* [x] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [x] I'm merging against the `development` branch
* [x] I ran `cargo-fmt --all` before pushing
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.

### CODE REVIEW

**Comment by CjS77** [comment]
Please remove Cargo.lock from VC and re-push

File: infrastructure/storage/src/lmdb.rs [hunk 1]
```diff
@@ -220,32 +230,57 @@ mod test {
 
     #[test]
     fn path_must_exist() {
+        let test_dir = "./tests/not_here/";
+        if std::fs::metadata(test_dir).is_ok() {
+            assert!(fs::remove_dir_all(test_dir).is_ok());
+        }
+        let msg;
+        if sys_info::os_type().unwrap().to_uppercase() == "WINDOWS" {
```
[the remark this answers is not in the document]
Comment (hansieodendaal) [reply infrastructure/storage/src/lmdb.rs]: Thanks, could not get this working.
Please test this in the MacOS, could only test Windows and Ubuntu Linux.

File: infrastructure/storage/src/lmdb.rs [hunk 2]
```diff
@@ -104,6 +104,14 @@ pub struct LMDBStore {
     pub(crate) curr_db: Arc<lmdb::Database<'static>>,
 }
 
+/// Consume self to remove all databases from scope
+impl LMDBStore {
+    fn delete_db_from_scope(self) -> Result<(), DatastoreError> {
+        self.curr_db.env().sync(true)?;
+        Ok(())
```
Comment (neonknight64) [inline infrastructure/storage/src/lmdb.rs]: Can probably replace ln 110 and 111 with only ```self.curr_db.env().sync(true)```, without the question mark and semicolon.

File: infrastructure/storage/src/lmdb.rs [hunk 3]
```diff
@@ -257,13 +290,22 @@ mod test {
         let val = store.get_raw(b"England").unwrap().unwrap();
         assert_eq!(str::from_utf8(&val).unwrap(), "rose");
         // Clean up
-        assert!(fs::remove_dir_all("./tests/test_default").is_ok());
+        assert!(store.delete_db_from_scope().is_ok());
+        let _no_val = fs::remove_dir_all(test_dir);
+        if std::fs::metadata(test_dir).is_ok() {
+            println!("Database file handles not released, still open in {:?}!", test_dir);
```
Comment (neonknight64) [inline infrastructure/storage/src/lmdb.rs]: Should ```println```not be removed from all the tests?

File: infrastructure/storage/src/lmdb.rs [hunk 4]
```diff
@@ -304,22 +355,28 @@ mod test {
...
     }
 
     #[test]
     fn read_and_write_10k_values() {
-        fs::create_dir("./tests/test_10k").unwrap();
+        let test_dir = "./tests/test_10k";
+        if std::fs::metadata(test_dir).is_ok() {
+            assert!(fs::remove_dir_all(test_dir).is_ok());
+        }
+        fs::create_dir(test_dir).unwrap();
```
Comment (neonknight64) [inline infrastructure/storage/src/lmdb.rs]: I think all the create directory calls can be replaced with ```assert!(fs::create_dir(test_dir).is_ok());```

File: infrastructure/storage/src/lmdb.rs (hunk 3 above)
Comment (hansieodendaal) [reply infrastructure/storage/src/lmdb.rs]: That is the only way to get some meaningful feedback during run-time if clean up dos not work, see below:

```
Testing started at 07:31 ...
Database file handles not released, still open in "./tests/test_default"!

thread 'lmdb::test::writes_to_default_db' panicked at 'assertion failed: fs::remove_dir_all(test_dir).is_ok()', infrastructure\storage\src\lmdb.rs:297:13
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

File: infrastructure/storage/src/lmdb.rs (hunk 2 above)
Comment (hansieodendaal) [reply infrastructure/storage/src/lmdb.rs]: Question mark can be removed yes, but semicolon is needed.

File: infrastructure/storage/src/lmdb.rs (hunk 4 above)
Comment (hansieodendaal) [reply infrastructure/storage/src/lmdb.rs]: Perfect

File: infrastructure/storage/src/lmdb.rs (hunk 2 above)
Comment (hansieodendaal) [reply infrastructure/storage/src/lmdb.rs]: Perfect

**Pull request merged**